tests/callbacks_test.go

```diff
@@ -312,22 +314,24 @@ func (s *CallbacksSuite) TestWorkflowNexusCallbacks_CarriedOver() {
 					err = nexus.HandlerErrorf(nexus.HandlerErrorTypeInternal, "intentional error")
 				}
 				ch.requestCompleteCh <- err
-				description, err := sdkClient.DescribeWorkflowExecution(ctx, request.WorkflowId, "")
-				s.NoError(err)
-				s.Equal(1, len(description.Callbacks))
-				callbackInfo := description.Callbacks[0]
-				s.ProtoEqual(request.CompletionCallbacks[0], callbackInfo.Callback)
-				s.ProtoEqual(&workflowpb.CallbackInfo_Trigger{Variant: &workflowpb.CallbackInfo_Trigger_WorkflowClosed{WorkflowClosed: &workflowpb.CallbackInfo_WorkflowClosed{}}}, callbackInfo.Trigger)
-				s.Equal(int32(attempt), callbackInfo.Attempt)
-				// Loose check to see that this is set.
-				s.Greater(callbackInfo.LastAttemptCompleteTime.AsTime(), time.Now().Add(-time.Hour))
-				if attempt < numAttempts {
-					s.Equal(enumspb.CALLBACK_STATE_BACKING_OFF, callbackInfo.State)
-					s.Equal("request failed with: 500 Internal Server Error", callbackInfo.LastAttemptFailure.Message)
-				} else {
-					s.Equal(enumspb.CALLBACK_STATE_SUCCEEDED, callbackInfo.State)
-					s.Nil(callbackInfo.LastAttemptFailure)
-				}
+				s.EventuallyWithT(func(col *assert.CollectT) {
+					description, err := sdkClient.DescribeWorkflowExecution(ctx, request.WorkflowId, "")
+					require.NoError(col, err)
+					require.Equal(col, 1, len(description.Callbacks))
+					callbackInfo := description.Callbacks[0]
+					protorequire.ProtoEqual(col, request.CompletionCallbacks[0], callbackInfo.Callback)
+					protorequire.ProtoEqual(col, &workflowpb.CallbackInfo_Trigger{Variant: &workflowpb.CallbackInfo_Trigger_WorkflowClosed{WorkflowClosed: &workflowpb.CallbackInfo_WorkflowClosed{}}}, callbackInfo.Trigger)
+					require.Equal(col, int32(attempt), callbackInfo.Attempt)
```

Comment:
This will still fail the test, you should change to assert and eager return.

```suggestion
					assert.Equal(col, int32(attempt), callbackInfo.Attempt)
					// Return to avoid evaluate the require statements below.
					if callbackInfo.Attempt != int32(attempt) {
						return
					}
```